lib/id3c/cli/command/etl/redcap_det.py

```diff
@@ -113,15 +121,23 @@ def decorated(*args, db: DatabaseSession, log_output: bool, fetch_batch_size: in
             api_url = urljoin(redcap_url, "api/")
             api_token = get_redcap_api_token(api_url)
 
+            if limit:
+                LOG.debug(f"Processing up to {limit:,} pending DETs")
+                limit_count = sql.Literal(limit)
+            else:
+                LOG.debug(f"Processing all pending DETs")
+                limit_count = sql.SQL("all")
+
             redcap_det = db.cursor(f"redcap-det {name}")
-            redcap_det.execute("""
+            redcap_det.execute(sql.SQL("""
                 select redcap_det_id as id, document
                   from receiving.redcap_det
                  where not processing_log @> %s
                    and document::jsonb @> %s
                  order by id
+                 limit {}
                    for update
-                """, (Json([etl_id]), Json(det_contains)))
+                """).format(limit_count), (Json([etl_id]), Json(det_contains)))
```

Comment:
Is is true that `redcap_det_id` is guaranteed to ascend over time?
I'm making that assumption based on code and comments below, just wanted to ask explicitly, as it seems pretty important to the logic here. If this is called with a limit option that is less than the number of pending DETs, then the `limit` in the query here, coupled with the `order by redcap_det_id`, should get us the earliest DETs which is what we want. 👍

Reply:
Yep. The ETLs intentionally process received records in insert order, and this isn't changed by the addition of `limit` here.

`redcap_det_id` is a generated id column controlled by this sequence:

```
tom@production.db.seattleflu.org/production=> \d receiving.redcap_det_redcap_det_id_seq 
          Sequence "receiving.redcap_det_redcap_det_id_seq"
  Type   │ Start │ Minimum │  Maximum   │ Increment │ Cycles? │ Cache 
─────────┼───────┼─────────┼────────────┼───────────┼─────────┼───────
 integer │     1 │       1 │ 2147483647 │         1 │ no      │     1
Sequence for identity column: receiving.redcap_det.redcap_det_id
```

If there are holes in the actually-used ids (due to aborted transactions or deleted rows), it would be possible to create new rows which would fill those with lower numbers than the sequence's current value. But in practice nothing is doing that.

Reply:
Excellent. Thanks Tom!